File: XMLCheck/Handler/CheckRulesXML15/Xml15RuleBase.cs
using XMLCheck.Models;
using XMLCheck.Services;

namespace XMLCheck.Handler.CheckRulesXML15
{
    public abstract class Xml15RuleBase:IXml15Rule
    {
        public abstract string Key { get; }

        public abstract ErrorDetails? Check(Xml15Model model,HoSoContext context);

        protected ErrorDetails Error(string message)
        {
            return new ErrorDetails
            {
                key = Key,
                message = message
            };
        }
    }
}

File: XMLValidator_v2/src/parsers/rules/xml15/xml15RuleBase.ts
import { Xml15Model } from '../../../interfaces/xml-models.interface';
import { HoSoContext } from '../../../services/hoso-context';
import { ErrorDetails } from '../../../dto/xml-check.dto';

export interface IXml15Rule {
  readonly key: string;
  check(model: Xml15Model, context: HoSoContext): ErrorDetails | null;
}

export abstract class Xml15RuleBase implements IXml15Rule {
  public abstract get key(): string;
  public abstract check(model: Xml15Model, context: HoSoContext): ErrorDetails | null;

  protected error(message: string): ErrorDetails {
    return {
      key: this.key,
      message,
    };
  }
}
